src/pages/Detail/Detail.js

```diff
@@ -1,9 +1,149 @@
 import React from 'react';
 
+import StoreImgBox from './StoreImgBox/StoreImgBox';
+import StoreInfo from './StoreInfo/StoreInfo';
+import StoreReviewBox from './StoreReviewBox/StoreReviewBox';
+
+import './Detail.scss';
+
 class Detail extends React.Component {
+  state = {
+    restaurants: null,
+    food: null,
+    image: null,
+    review: [],
+  };
+
+  reviewRequestNum = 1;
+  reviewRequestNumLimit; //한 번에 10개씩 띄울 경우 총 리뷰 페이지 수
+
+  restaurantsAddr = `restaurants/${this.props.match.params.id}`;
+  foodsAddr = `restaurants/${this.props.match.params.id}/food`;
+  imgAddr = `restaurants/${this.props.match.params.id}/food/image`;
+  reviewsAddr = `restaurants/${this.props.match.params.id}/review?limit=${this.reviewRequestNum}`;
+
+  componentDidMount = () => {
+    this.fetchData(this.restaurantsAddr);
+    this.fetchData(this.foodsAddr);
+    this.fetchData(this.imgAddr);
+    this.fetchReviewData(1, 5);
+  };
+
+  setRequestNumLimit = () => {
+    const { restaurants } = this.state;
+
+    restaurants.review_count.total % 10 === 0
+      ? (this.reviewRequestNumLimit = restaurants.review_count.total / 10)
+      : (this.reviewRequestNumLimit =
+          Math.floor(restaurants.review_count.total / 10) + 1);
+  };
+
+  handleReviewEdit = () => {
+    console.log('edit');
+  };
+
+  handleReviewDel = reviewId => {
+    fetch(
+      `http://${IP_ADDRESS}:8000/restaurants/${this.props.match.params.id}/review`,
+      {
+        method: 'DELETE',
+        body: JSON.stringify({
+          review_id: reviewId,
+        }),
+      }
+    ).then(() => this.reFetchReviewData(1, 5));
+  };
+
+  sortAddr = addr => {
+    const splitAddr = addr.split('/');
+
+    return parseInt(splitAddr[splitAddr.length - 1]) ==
+      this.props.match.params.id
+      ? splitAddr[0]
+      : splitAddr[splitAddr.length - 1];
+  };
+
+  fetchReviewData = (min, max) => {
+    fetch(
+      `http://${IP_ADDRESS}:8000/restaurants/${this.props.match.params.id}/review?limit=${this.reviewRequestNum}&rating-min=${min}&rating-max=${max}`,
+      {
+        method: 'GET',
+      }
+    )
+      .then(res => res.json())
+      .then(res => {
+        this.setState({ review: [...this.state.review, ...res.result] });
```

Comment:
```suggestion
        this.setState(prev => ({ review: [...prev, ...res.result] }));
```
가급적 외부 요소에 대한 의존을 줄이는 방식으로 코딩하는 것이 좋기 때문에 이전 값을 참조하는 방식으로 고치면 좋겠습니다